### ISSUE TITLE
Should 404 extend from base or main ?

### ISSUE THREAD

**Issue by AbdealiJK** [squidfunk/mkdocs-material#1865]
__I checked that...__

- [x] ... the documentation does not mention anything about my problem
- [ ] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... there are no open or closed issues that are related to my problem

## Description
I was using this theme for one of my documentation projects, and i am doing theme customizations for my docs.
I realized that the theme customizations I am adding are not being applied to the 404 page.

I followed the documentation note:
> To override a block, create a main.html inside the overrides directory and define the block, e.g.: ...
I created my own `main.html` and added some styles there

NOTE: I'm aware of the extra_css and extra_javascript that mkdocs provides ... but I didn't use them for other reasons

### Expected behavior
I was expecting the CSS styles I defined to be applied to all pages.

### Actual behavior
Things worked quite well for doc pages. But the css styles I had did not get applied to the 404 page.

On digging into github here, I saw that 404.html extends `base.html` and not `main.html` - https://github.com/squidfunk/mkdocs-material/blob/master/src/404.html#L23
Which is probably what is causing my issue, as I had overridden the `main.html` - not `base.html`

### Steps to reproduce the bug
My `main.html`:
```jinja2
{% extends "base.html" %}

{% block styles %}
    {{ super() }}
    <link rel="stylesheet" href="{{ 'assets/stylesheets/extra.css' | url }}" />
{% endblock %}
```

### Package versions
* Python: Python 3.7.7
* MkDocs: mkdocs, version 1.1.2
* Material: Version: 5.4.0

### Project configuration
My mkdocs.yml:
```yaml
theme:
  name: material
  custom_dir: theme
```

### System information
* OS: Windows 10 + Ubuntu 18.04 WSL
* Browser: Google chrome 84.0

**Comment by squidfunk** [comment]
Thanks for reporting. That's a valid concern. We implemented it like this because MkDocs does the same for their templates. However, we can check whether we could just extend it from `main.html`, as I also think it would be better if all customizations would be picked up.

**Comment by squidfunk** [comment]
Fixed in e3d79b3b. It's funny that you're the first one to realize this inconsistency, but maybe it's related to the fact that most people don't customize the theme. My testing shows that there should be no problems or unintended side effects.

Good to go!

**Comment by squidfunk** [comment]
Released as part of 5.5.6.

**Comment by AbdealiJK** [comment]
=D thanks a lot for the super swift reply and fix !
